reports-backend/src/main/java/org/jboss/da/reports/api/AdvancedArtifactReport.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    public void advancedReportGeneration() {
+        Set<GAV> modulesAnalyzed = new HashSet<>();
+        populateAdvancedFields(artifactReport, modulesAnalyzed);
+    }
+
+    /**
+     * The logic on how the advanced fields are populated is determined from
+     * DA-204
+     *
+     * @param report          report to analyze
+     * @param modulesAnalyzed set of modules already analyzed
+     */
+    private void populateAdvancedFields(ArtifactReport report, Set<GAV> modulesAnalyzed) {
+
+        for (ArtifactReport dep : report.getDependencies()) {
+            if (modulesAnalyzed.contains(dep.getGav())) {
+                // if module already analyzed, skip
+                continue;
+            } else if (isDependencyAModule(dep)) {
+                // if dependency is a module, but not yet analyzed
+                modulesAnalyzed.add(dep.getGav());
+                populateAdvancedFields(dep, modulesAnalyzed);
+            } else {
+                // only generate populate advanced report with community GAVs
+                if (VersionParser.isRedhatVersion(dep.getVersion())) continue;
+
+                // we have a top-level module dependency
+                if (dep.isWhitelisted()) whitelistArtifacts.add(dep.getGav());
+                if (dep.isBlacklisted()) blacklistArtifacts.add(dep.getGav());
+
+                if (dep.getBestMatchVersion().isPresent()) {
+                    communityGavsWithBestMatchVersions.add(dep.getGav());
+                    // also add bestMatchVersion GAV to set
+                    communityGavsWithBestMatchVersions.add(new GAV(dep.getGroupId(),
+                            dep.getArtifactId(), dep.getBestMatchVersion().get()));
+                } else if (!dep.getAvailableVersions().isEmpty()) {
+                    communityGavsWithBuiltVersions.add(dep.getGav());
+
+                    // also add available versions to set
+                    dep.getAvailableVersions().stream()
+                            .forEach(v -> communityGavsWithBuiltVersions
+                                    .add(new GAV(dep.getGroupId(), dep.getArtifactId(), v)));
+
+                    // add any whitelisted GAVs, if any
+                    addWhitelistedGAVToCommunityGavsWithBuiltVersions(dep.getGroupId(), dep.getArtifactId());
+                } else if (hasWhitelistedGA(dep.getGroupId(), dep.getArtifactId())) {
+                    // no bestmatch version, no built versions, but we have GAVs whitelisted
+                    addWhitelistedGAVToCommunityGavsWithBuiltVersions(dep.getGroupId(), dep.getArtifactId());
+                } else {
+                    communityGavs.add(dep.getGav());
+                }
+            }
+        }
+    }
+
+    private boolean isDependencyAModule(ArtifactReport dependency) {
+        return dependency.getGroupId().equals(artifactReport.getGroupId())
+                && dependency.getVersion().equals(artifactReport.getVersion());
```

Comment:
We should decide if an artifact is module by looking into the SCM repository, not by comparsion of groupid/version.